Assets/Mirror/Runtime/Transport/ClientTransport.cs

```diff
@@ -0,0 +1,136 @@
+        /// <para>Example usage: return Application.platform == RuntimePlatform.WebGLPlayer</para>
+        /// </summary>
+        /// <returns>True if this transport works in the current platform</returns>
+        public abstract bool Available();
+
+        /// <summary>
+        /// Notify subscribers when when this client establish a successful connection to the server
+        /// <para>callback()</para>
+        /// </summary>
+        public Action OnConnected = () => Debug.LogWarning("OnConnected called with no handler");
+
+        /// <summary>
+        /// Notify subscribers when this client receive data from the server
+        /// <para>callback(ArraySegment&lt;byte&gt; data, int channel)</para>
+        /// </summary>
+        public Action<ArraySegment<byte>, int> OnDataReceived = (data, channel) => Debug.LogWarning("OnDataReceived called with no handler");
+
+        /// <summary>
+        /// Notify subscribers when this client encounters an error communicating with the server
+        /// <para>callback(Exception e)</para>
+        /// </summary>
+        public Action<Exception> OnError = (error) => Debug.LogWarning("OnError called with no handler");
+
+        /// <summary>
+        /// Notify subscribers when this client disconnects from the server
+        /// <para>callback()</para>
+        /// </summary>
+        public Action OnDisconnected = () => Debug.LogWarning("OnDisconnected called with no handler");
+
+        /// <summary>
+        /// Determines if we are currently connected to the server
+        /// </summary>
+        /// <returns>True if a connection has been established to the server</returns>
+        public abstract bool Connected();
+
+        /// <summary>
+        /// Establish a connection to a server
+        /// </summary>
+        /// <param name="address">The IP address or FQDN of the server we are trying to connect to</param>
+        public abstract void Connect(string address);
+
+        /// <summary>
+        /// Establish a connection to a server
+        /// </summary>
+        /// <param name="uri">The address of the server we are trying to connect to</param>
+        public virtual void Connect(Uri uri)
+        {
+            // By default, to keep backwards compatibility, just connect to the host
+            // in the uri
+            Connect(uri.Host);
+        }
+
+        /// <summary>
+        /// Send data to the server
+        /// </summary>
+        /// <param name="channelId">The channel to use.  0 is the default channel,
+        /// but some transports might want to provide unreliable, encrypted, compressed, or any other feature
+        /// as new channels</param>
+        /// <param name="segment">The data to send to the server. Will be recycled after returning, so either use it directly or copy it internally. This allows for allocation-free sends!</param>
+        public abstract void ClientSend(int channelId, ArraySegment<byte> segment);
```

Comment:
This is the client class,  so ClientSend is redundant
```suggestion
        public abstract void Send(int channelId, ArraySegment<byte> segment);
```